modules/apps/portal-template/portal-template-engine-api/src/main/java/com/liferay/portal/template/engine/TemplateContextHelper.java

```diff
@@ -193,6 +203,16 @@ public Set<String> getRestrictedVariables() {
 		return Collections.emptySet();
 	}
 
+	public List<TemplateContextContributor> getTemplateContextContributors() {
```

Comment:
@JanisZhang Please make this method protected, and return `Iterable<>`, so that you can directly return the `ServiceTrackerList`.